components/brave_wallet_ui/utils/amount.ts

```diff
@@ -0,0 +1,289 @@
+import BigNumber from 'bignumber.js'
+
+import { CurrencySymbols } from './currency-symbols'
+
+type BigNumberIsh =
+  | BigNumber
+  | string
+  | number
+
+type AmountLike =
+  | Amount
+  | BigNumberIsh
+
+export default class Amount {
+  public readonly value?: BigNumber
+
+  public constructor (value: BigNumberIsh) {
+    this.value = value === ''
+      ? undefined
+      : new BigNumber(value)
+  }
+
+  static zero (): Amount {
+    return new Amount('0')
+  }
+
+  static empty (): Amount {
+    return new Amount('')
+  }
+
+  plus (value: AmountLike): Amount {
+    if (value instanceof Amount) {
+      return this.plus(value.value || '')
+    }
+
+    if (value === '') {
+      return this
+    } else if (this.value === undefined) {
+      return new Amount(value)
+    }
+
+    return new Amount(this.value.plus(value))
+  }
+
+  times (value: AmountLike): Amount {
+    if (value instanceof Amount) {
+      return this.times(value.value || '')
+    }
+
+    if (value === '' || this.value === undefined) {
+      return Amount.empty()
+    }
+
+    return new Amount(this.value.times(value))
+  }
+
+  div (value: AmountLike): Amount {
```

Comment:
Can we add a check if the value is 0 to avoid getting `Infinity` as a result of division by 0?

Reply:
Fixed and added a test.